src/android/CodePushReportingManager.java

```diff
@@ -13,28 +12,7 @@
 
     private Activity cordovaActivity;
     private CodePushPreferences codePushPreferences;
-
-    /**
-     * Defines application statuses we use in reporting events from the native to the JS layer.
-     */
-    public static enum Status {
-        STORE_VERSION(0),
-        UPDATE_CONFIRMED(1),
-        UPDATE_ROLLED_BACK(2);
-
-        private int value;
-
-        Status(int i) {
-            this.value = i;
-        }
-
-        /**
-         * Returns the value associated with this enum.
-         */
-        public int getValue() {
-            return this.value;
-        }
-    }
+    private Boolean hasFailedReport = null;
```

Comment:
oh I see, you're using a different shared preferences for each preference...I don't think this is necessary, I think you can put all the preferences in a single shared preference and it will be faster, but given that all the preferences are separate, this storing this variable like this is definitely faster.